src/commands/UploadReactNativeCommand.ts

```diff
@@ -213,83 +207,35 @@ function marshallVersion(platform: Platform, opts: Record<string, unknown>): Ver
       throw new Error('--app-version-code and --code-bundle-id cannot both be given')
     }
 
-    if (typeof opts.codeBundleId !== 'string') {
-      throw new Error('--code-bundle-id must be a string')
-    }
-
-    return {
-      type: VersionType.CodeBundleId,
-      codeBundleId: opts.codeBundleId,
-    }
+    return
   }
 
   throw new Error('Either --app-version or --code-bundle-id must be given')
 }
 
-function marshallPlatformOptions(platform: Platform, version: Version, opts: Record<string, unknown>): PlatformOptions {
-  switch (platform) {
-    case Platform.Ios: {
-      if (opts.appVersionCode) {
-        throw new Error('--app-version-code cannot be given with --platform "ios"')
-      }
-
-      let appBundleVersion
-
-      if (opts.appBundleVersion && typeof opts.appBundleVersion === 'string') {
-        appBundleVersion = opts.appBundleVersion
-      }
-
-      return { type: platform, appBundleVersion }
+function validatePlatformOptions(opts: Record<string, unknown>): void {
+  switch (opts.platform) {
+    case 'ios': {
+      if (opts.appVersionCode) throw new Error('--app-version-code cannot be given with --platform "ios"')
+      break
     }
-
-    case Platform.Android: {
-      if (opts.appBundleVersion) {
-        throw new Error('--app-bundle-version cannot be given with --platform "android"')
-      }
-
-      let appVersionCode
-
-      if (opts.appVersionCode && typeof opts.appVersionCode === 'string') {
-        appVersionCode = opts.appVersionCode
-      }
-
-      return { type: platform, appVersionCode }
+    case 'android': {
+      if (opts.appBundleVersion) throw new Error('--app-bundle-version cannot be given with --platform "android"')
+      break
     }
   }
 }
 
-function marshallRetrieval(opts: Record<string, unknown>): SourceMapRetrieval {
-  if (opts.fetch) {
-    let url = 'http://localhost:8081'
-
-    if (opts.bundlerUrl && typeof opts.bundlerUrl === 'string') {
-      url = opts.bundlerUrl
+function validateRetrieval(opts: Record<string, unknown>): void {
+  if (!opts.fetch){
+    if (!opts.sourceMap || typeof opts.sourceMap !== 'string') {
```

Comment:
These `typeof` checks were just for making Typescript happy; they're probably not required anymore, though I guess there's no harm in keeping them

Reply:
They aren't necessary for the logic in that function. We're not guaranteed by the type system that these values are strings, but we are guaranteed by the parsing logic in `command-line-args` so that's good enough for me.